scripts/rook.sh

```diff
@@ -84,29 +83,19 @@ kubectl_retry() {
 function deploy_rook() {
 	kubectl_retry create -f "${ROOK_URL}/common.yaml"
 	kubectl_retry create -f "${ROOK_URL}/operator.yaml"
-	# find out the rook version to decide on the ceph cluster image to be used
-	ROOK_MAJOR=$(rook_version 1)
-	ROOK_MINOR=$(rook_version 2)
-	if { [ "${ROOK_MAJOR}" -eq 1 ] && [ "${ROOK_MINOR}" -le 2 ]; }; then
-		ROOK_CEPH_CLUSTER_VERSION_IMAGE_PATH="image: ceph/ceph:${ROOK_CEPH_CLUSTER_VERSION}"
-		# upgrade ceph cluster version to 14.2.10 to support CephFS snapshot functionalities.
-		TEMP_DIR="$(mktemp -d)"
-		curl -o "${TEMP_DIR}"/cluster-test.yaml "${ROOK_URL}/cluster-test.yaml"
-		sed -i "s|image.*|${ROOK_CEPH_CLUSTER_VERSION_IMAGE_PATH}|g" "${TEMP_DIR}"/cluster-test.yaml
-		cat  "${TEMP_DIR}"/cluster-test.yaml
-		kubectl_retry create -f "${TEMP_DIR}/cluster-test.yaml"
-		rm -rf "${TEMP_DIR}"
-	else
-		# add "mon_warn_on_pool_no_redundancy = false" to ceph.conf if missing
-		# see https://github.com/rook/rook/pull/5925 for upstream status
-		TEMP_DIR="$(mktemp -d)"
-		curl -o "${TEMP_DIR}"/cluster-test.yaml "${ROOK_URL}/cluster-test.yaml"
-		if ! grep -q mon_warn_on_pool_no_redundancy "${TEMP_DIR}"/cluster-test.yaml; then
-			sed -i '/osd_pool_default_size =/a \    mon_warn_on_pool_no_redundancy = false' "${TEMP_DIR}"/cluster-test.yaml
-		fi
-		kubectl_retry create -f "${TEMP_DIR}/cluster-test.yaml"
-		rm -rf "${TEMP_DIR}"
-	fi
+    # Override the ceph version which rook installs by default.
+    if [ -z "${ROOK_CEPH_CLUSTER_VERSION}" ]
+    then
+        kubectl_retry create -f "${ROOK_URL}/cluster-test.yaml"
+    else
+        ROOK_CEPH_CLUSTER_VERSION_IMAGE_PATH="image: ceph/ceph:${ROOK_CEPH_CLUSTER_VERSION}"
+	    TEMP_DIR="$(mktemp -d)"
```

Comment:
now there is a difference between the 1st line in this else block, compared to the next lines?

Reply:
sorry for repeating same mistake again, bad day with keyboard today :(